src/os/filestore/FileStore.cc

```diff
@@ -472,13 +472,7 @@ int FileStore::lfn_unlink(const coll_t& cid, const ghobject_t& o,
     }
 
     if (!force_clear_omap) {
-      if (hardlink == 0) {
-          if (!m_disable_wbthrottle) {
-	    wbthrottle.clear_object(o); // should be only non-cache ref
-          }
-	  fdcache.clear(o);
-	  return 0;
-      } else if (hardlink == 1) {
+      if (hardlink == 0 || hardlink == 1) {
```

Comment:
So to clarify the situation is

 - clear omap
 - unlink
 - unlink persists to disk, but omap does not
 - <crash>
 - replay sees hardlink 0 and doesn't clear omap

?